### ISSUE TITLE
python-language-server replacement

### ISSUE THREAD

**Issue by arthurio** [dense-analysis/ale#3722]
**Name:** python-lsp-server
**URL:** https://github.com/python-lsp/python-lsp-server

It seems that `python-language-server` is not longer maintained (I asked for confirmation [here](https://github.com/palantir/python-language-server/issues/935)). I suggest updating the documentation to point to `python-lsp-server` instead, which is the new actively maintained fork of the project.

In terms of configuration, not much needed to be changed:

### Initial config

```vimrc
# .vimrc
let g:ale_linters = {
\   'python': ['flake8', 'pyls', 'bandit', 'mypy'],
\}

let g:ale_python_pyls_config = {
\   'pyls': {
\     'plugins': {
\       'pycodestyle': {
\         'enabled': v:false,
\       },
\       'pyflakes': {
\         'enabled': v:false,
\       },
\       'pydocstyle': {
\         'enabled': v:false,
\       },
\     },
\   },
\}
```

### New config

```vimrc
# .vimrc
let g:ale_linters = {
\   'python': ['flake8', 'pyls', 'bandit', 'mypy'],
\}

let g:ale_python_pyls_executable = "pylsp"

let g:ale_python_pyls_config = {
\   'pylsp': {
\     'plugins': {
\       'pycodestyle': {
\         'enabled': v:false,
\       },
\       'pyflakes': {
\         'enabled': v:false,
\       },
\       'pydocstyle': {
\         'enabled': v:false,
\       },
\     },
\   },
\}
```

To recap, just set `let g:ale_python_pyls_executable` to `"pylsp"` (previously `pyls`), and update `pyls` to `pylsp` in `g:ale_python_pyls_config` if you need to configure it.

**Comment by jtuz** [comment]
Thank you @arthurio  this was helpful for me

**Comment by mkalinski** [comment]
Thanks. Unfortunately, this doesn't seem to work with `pylsp` installed in a virtualenv. It seems that when setting `ale_python_*_executable`, ALE expects the executable to be global.

I'm not sure if I'm missing something, or overriding a local executable's name is something ALE just doesn't support.

**Comment by arthurio** [comment]
@mkalinski I use virtualenvs and things work fine for me (I use pyenv). I would check the following:
1. Run `which pylsp` from the folder you launch vim
2. Can you run other python executables from vim? (does your vim has python support?)
3. Run `:!which python` from within vim

For me:
```
which pylsp
/Users/arthur/.pyenv/shims/pylsp
```

```
# in vim
:!which python

/Users/arthur/.pyenv/shims/python

Press ENTER or type command to continue
```

I hope this helps.

**Comment by mkalinski** [comment]
This isn't really my case. I'm using ALE's functionality to automatically use virtualenvs.

When there's a virtualenv named `venv` or `.venv` in the root directory of a python project, ALE will enter it to run any configured linters and fixers from there. It has nothing to do with which virtualenv (if any) vim runs in.

This works for other tools I have installed in the virtualenv, that don't have changed executable names, ex. flake8. And I can run pylsp when I enter the virtualenv manually. But after setting `g:ale_python_pyls_executable` ALE refuses to use this automatic-enter functionality for pylsp.

**Comment by jtuz** [comment]
@mkalinski, if you are using `g:ale_virtualenv_dir_names`, according to ALE documentation it only look upwards the virtual directory in your root project for all available tools, but it doesn't load your virtualenv like a  `pyenv activate venv` or  `source venv/bin/activate` , so to take effect the configuration above made by @arthurio ,  I suggest two options to prove:

1. set `g:ale_python_pyls_executable` as an absolute path (this can be annoying if you have many projects, you need to chance the path each time you work on your projects)
2. the simplest way, before you open vim or nvim, load your virtualenv whatever your are using `pyenv` or `venv` module from python, this will let your .vimrc clean and compatible among your projects

**Comment by GerardoGR** [comment]
Thank you @arthurio for the sample configuration. Did you manage to have the autocompletion working?. Mine is not working for some reason :(

**Comment by arthurio** [comment]
@GerardoGR Yes autocompletion is working for me 🤷🏻

**Comment by GerardoGR** [comment]
To answer my own question and in case someone else experience the same. I had `set paste` in my `vimrc` config. According to the docs:

> ALE automatic completion will not work when 'paste' is active. Only set
'paste' when you are copy and pasting text into your buffers.
[source](https://github.com/dense-analysis/ale/blob/master/doc/ale.txt#L478)

When I disabled paste (removing it from my `vimrc`) the auto-completion worked as expected :slightly_smiling_face:
